session/sessiondata_test.go

```diff
@@ -0,0 +1,259 @@
+				})
+			})
+		})
+	})
+}
+
+// TestGetOauth2TokenUserNotSignedIn verifies that nothing is returned when
+// a user is signed in
+func TestGetOauth2TokenNotUserSignedIn(t *testing.T) {
+
+	Convey("Given I have session data for a non-signed-in session", t, func() {
+
+		var sessionData SessionData = map[string]interface{}{
+			"signin_info": map[string]interface{}{
+				"signed_in": int8(0),
+			},
+		}
+
+		Convey("When I call GetOauth2Token", func() {
+
+			tok := sessionData.GetOauth2Token()
+
+			Convey("Then nothing should be retuned", func() {
+
+				So(tok, ShouldBeNil)
+			})
+		})
+	})
+}
+
+// TestIsSignedInEmptySessionDataMap verifies that false is returned when
+// checking if an empty session is signed in
+func TestIsSignedInEmptySessionDataMap(t *testing.T) {
+
+	Convey("Given I have an empty session map", t, func() {
+
+		var sessionData SessionData = map[string]interface{}{}
+
+		Convey("When I call isSignedIn", func() {
+
+			signedIn := sessionData.isSignedIn()
+
+			Convey("Then I should return false", func() {
+
+				So(signedIn, ShouldBeFalse)
+			})
+		})
+	})
+}
+
+// TestGetExpirationHappyPath verifies that expiration is returned successfully
+func TestGetExpirationHappyPath(t *testing.T) {
+
+	Convey("Given I have some session data with an 'expires_in' token", t, func() {
+
+		expiresIn := uint16(123)
+
+		var sessionData SessionData = map[string]interface{}{
+			"signin_info": map[string]interface{}{
+				"access_token": map[string]interface{}{
+					"expires_in": expiresIn,
+				},
+			},
+		}
+
+		Convey("When I call GetExpiration", func() {
+
+			expiration := sessionData.GetExpiration()
+
+			Convey("Then expiration should be returned", func() {
+
+				So(expiration, ShouldEqual, uint64(expiresIn))
+			})
+		})
+	})
+}
+
+// TestGetExpirationNonePresent verifies that when expiration is not present on
+// the session, 0 is returned
+func TestGetExpirationNonePresent(t *testing.T) {
+
+	Convey("Given I have some session data with no 'expires_in' token", t, func() {
+
+		var sessionData SessionData = map[string]interface{}{
+			"signin_info": map[string]interface{}{
+				"access_token": map[string]interface{}{},
+			},
+		}
+
+		Convey("When I call GetExpiration", func() {
+
+			expiration := sessionData.GetExpiration()
+
+			Convey("Then 0 should be returned", func() {
+
+				So(expiration, ShouldEqual, uint64(0))
+			})
+		})
+	})
+}
+
+// TestRefreshExpiration verifies that once refreshed, expiration is not nil
+func TestRefreshExpiration(t *testing.T) {
+
+	Convey("Given I have some session data", t, func() {
+
+		var sessionData SessionData = map[string]interface{}{
+			"signin_info": map[string]interface{}{
+				"access_token": map[string]interface{}{},
+			},
+		}
+
+		Convey("When I call RefreshExpiration", func() {
+
+			sessionData.RefreshExpiration()
+
+			Convey("Then 'expires' should be set", func() {
+
+				So(sessionData.getExpiry(), ShouldNotBeNil)
```

Comment:
Shouldn't this test verify what the expiration is set to?

Reply:
We can't really test this effectively because expiry depends on the current time. We could test that the time falls within bounds, but this could then fail sporadically if tests run slow for any reason.